tests/lib/nm/linux_bridge_port_vlan_test.py

```diff
@@ -0,0 +1,255 @@
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: 1000,
+                }
+            },
+        ],
+    ],
+    ids=[
+        "no-trunk-tags",
+        "single-trunk-tag",
+        "two-single-trunk-tags",
+        "tag-ranges",
+        "multiple-tag-ranges",
+        "mixed-single-tag-and-tag-ranges",
+    ],
+)
+@pytest.mark.parametrize(
+    "is_native_vlan", [False, True], ids=["not-native-vlan", "native-vlan"]
+)
+def test_generate_nm_vlan_filtering_config(trunk_tags, is_native_vlan):
+    port_vlan_filter = PortVlanFilter()
+    port_vlan_filter.create_configuration(
+        trunk_tags, ACCESS_TAG if is_native_vlan else None, is_native_vlan
+    )
+    port_bridge_vlan_list = port_vlan_filter.to_nm()
+    expected_trunk_tag_length = len(trunk_tags) + (1 if is_native_vlan else 0)
+    assert expected_trunk_tag_length == len(port_bridge_vlan_list)
+    assert is_native_vlan == any(
+        [
+            port_bridge_vlan.is_pvid() and port_bridge_vlan.is_untagged()
+            for port_bridge_vlan in port_bridge_vlan_list
+        ]
+    )
+    desired_trunk_tags = _port_vlan_trunk_tags_to_tag_list(
+        trunk_tags, ACCESS_TAG, is_native_vlan
+    )
+    _assert_port_settings_vlan_conf(port_bridge_vlan_list, desired_trunk_tags)
+
+
+@pytest.mark.parametrize(
+    "trunk_tags",
+    [
+        [],
+        [100],
+        [100, 200],
+        [(100, 200)],
+        [100, (200, 300)],
+        [(100, 200), (300, 400)],
+    ],
+    ids=[
+        "access-port",
+        "single-trunk-tag",
+        "two-trunk-tags",
+        "one-tag-range",
+        "mixed-tag-and-range",
+        "multiple-ranges",
+    ],
+)
+@pytest.mark.parametrize(
+    "is_native_vlan", [False, True], ids=["not-native-vlan", "native-vlan"]
+)
+def test_bridge_port_vlan_to_dict(NM_mock, trunk_tags, is_native_vlan):
+    port_vlans = _generate_port_vlan_mocks(
+        trunk_tags, is_native_vlan, ACCESS_TAG
+    )
+
+    vlan_config = PortVlanFilter()
+    vlan_config.import_from_bridge_settings(port_vlans)
+    assert vlan_config.tag == (
+        ACCESS_TAG if (not trunk_tags or is_native_vlan) else None
+    )
+    assert len(vlan_config.trunk_ports) == len(trunk_tags)
+    assert vlan_config.is_native_vlan_enabled == (
+        is_native_vlan and len(trunk_tags) > 0
+    )
+    assert vlan_config.to_dict() == _get_vlan_config_dict(
+        trunk_tags, ACCESS_TAG, is_native_vlan
+    )
+
+
+def _port_vlan_trunk_tags_to_tag_list(trunk_tags, access_tag, enable_vlan):
+    tag_list = []
+    for trunk_tag in trunk_tags:
+        single_tag = trunk_tag.get(LB.Port.Vlan.TrunkTags.ID)
+        range_tags = trunk_tag.get(LB.Port.Vlan.TrunkTags.ID_RANGE, {})
+        if single_tag:
+            tag_list.append(single_tag)
+        else:
+            for tag in range(
+                range_tags.get(LB.Port.Vlan.TrunkTags.MIN_RANGE),
+                range_tags.get(LB.Port.Vlan.TrunkTags.MAX_RANGE) + 1,
+            ):
+                tag_list.append(tag)
```

Comment:
I do not like the fact that structures are created like this, but if we are at it then:
```
tag_list += range(range_tags.get(LB.Port.Vlan.TrunkTags.MIN_RANGE), range_tags.get(LB.Port.Vlan.TrunkTags.MAX_RANGE))
```